R/modify_pd.R

```diff
@@ -9,33 +9,27 @@ NULL
 
 #' @describeIn update_indention Inserts indetion based on round brackets.
 indent_round <- function(pd, indent_by) {
-  indention_needed <- needs_indention(pd, token = "'('")
+  opening <- which(pd$token == "'('")
+  indention_needed <- needs_indention(pd, token = "'('", opening[1])
   if (indention_needed) {
-    opening <- which(pd$token == "'('")
     start <- opening + 1
     stop <- nrow(pd) - 1
-    if (start > stop) return(pd)
-
-    pd <- pd %>%
-      mutate(indent = indent + ifelse(seq_len(nrow(pd)) %in% start:stop,
-                                      indent_by, 0))
+    pd$indent <- pd$indent +
+      ifelse(between(seq_len(nrow(pd)), start, stop), indent_by, 0)
   }
 
   pd %>%
     set_unindention_child(token = "')'", unindent_by = indent_by)
 }
 #' @rdname update_indention
 indent_curly <- function(pd, indent_by) {
-  indention_needed <- needs_indention(pd, token = "'{'")
+  opening <- which(pd$token == "'{'")
+  indention_needed <- needs_indention(pd, token = "'{'", opening[1])
```

Comment:
Ok no, I just figured out that `which()` is not expensive. It was the comparison inside `which()`, which we can't avoid anyways. Also, always having a numerical vector is better than having NULL sometimes, so I use indices anyways.